### PR TITLE
Add J2CL attachment composer controller

### PR DESCRIPTION
**Pull request by vega113**
Part of #971. Implements Task 4 from `docs/superpowers/plans/2026-04-23-issue-971-rich-edit-attachments.md`.

Summary:
- Adds `J2clAttachmentComposerController` for file and pasted-image attachment selection, queueing, progress, cancellation/reset, and insertion callbacks.
- Adds `J2clAttachmentIdGenerator` with legacy-compatible `domain/seedToken` ids and validation.
- Keeps this slice UI-framework-neutral; Lit wiring remains a later task.
- Releases original file/blob payload references once an upload reaches a terminal or stale-completion path while preserving visible queue metadata.

Verification:
- `cd j2cl && ./mvnw -Dtest=J2clAttachmentComposerControllerTest,J2clAttachmentIdGeneratorTest,J2clAttachmentUploadClientTest,J2clAttachmentMetadataClientTest,J2clRichContentDeltaFactoryTest,J2clPlainTextDeltaFactoryTest test` — 100 tests, 0 failures.
- `cd j2cl && ./mvnw test` — 453 tests, 0 failures, 61 skipped.
- `git diff --check origin/main..HEAD` — passed.

Review:
- Self-review completed during the round-5 through round-12 fix loop.
- Claude review round 12 approved with no blockers.
- Final review output: `/tmp/issue-971-task4-attachment-composer-claude-round12.out`.

## Summary by CodeRabbit

* **New Features**
  * Attachment uploads in composer (file select + image paste) processed sequentially with automatic start, insertion into composer, and deterministic attachment IDs.
  * Captions default to trimmed filenames when blank; display-size options supported.

* **Bug Fixes**
  * Progress clamped to 0–100% and late callbacks ignored after cancel/reset.
  * Per-item error reporting: upload/insert failures marked and queue continues.

* **Tests**
  * Added comprehensive end-to-end tests covering enqueueing, progress, paste uploads, insertion, validation, cancel/reset, and edge cases.

### CODE REVIEW

File: j2cl/src/main/java/org/waveprotocol/box/j2cl/attachment/J2clAttachmentComposerController.java [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+          item.attachmentId, waveRef, item.payload, progressCallback, uploadCallback);
+    } else {
+      uploadClient.uploadFile(
+          item.attachmentId,
+          waveRef,
+          item.payload,
+          item.fileName,
+          progressCallback,
+          uploadCallback);
+    }
```
[the remark this answers is not in the document]
Comment (vega113) [reply j2cl/src/main/java/org/waveprotocol/box/j2cl/attachment/J2clAttachmentComposerController.java:338]: Fixed in `6ea9179ac` by wrapping `uploadClient.uploadFile(...)` and `uploadClient.uploadPastedImage(...)` in a `try/catch (RuntimeException)`, routing synchronous upload-start failures through `handleUploadComplete(...)` as `FAILED`/`NETWORK`, clearing `uploadInProgress`, releasing the payload via the existing terminal path, and continuing the queue. Added regression coverage for file and pasted-image synchronous start failures; local verification: focused J2CL 102 tests / 0 failures, full J2CL 455 tests / 0 failures / 61 skipped.
